src/Main.cc

```diff
@@ -333,6 +333,18 @@ static int tclReadlineInit(Tcl_Interp* interp)
 }
 #endif
 
+int tclEvalFile(const char* filename, Tcl_Interp* interp)
+{
+  int code = Tcl_EvalFile(interp, filename);
+  if (code != TCL_OK) {
+    const char* result = Tcl_GetStringResult(interp);
+    if (result[0] != '\0') {
+      printf("%s\n", result);
```

Comment:
Won't tcl print the result itself?

Reply:
I was trying not to change this aspect of the existing behavior. I will check

Reply:
Ah. There more to unravel here...

There's substantial amount of useful error context that isn't printed when using sta::sourceTclFile compared to  tclsh, like a stack trace.

If I disable the monkeypatched "source" implementation in sta and modify tclEvalFile() to:

```C++
int tclEvalFile(const char* filename, Tcl_Interp* interp)
{
  int code = Tcl_EvalFile(interp, filename);
  if (code != TCL_OK) {
    const char* errorMessage = Tcl_GetStringResult(interp);
    const char* errorStack = Tcl_GetVar(interp, "errorInfo", TCL_GLOBAL_ONLY);

    printf("Error: %s\n", errorMessage);
    if (errorStack != NULL) {
        printf("Stack Trace:\n%s\n", errorStack);
    }
  }
  return code;
}
```

I get:

```
$ tclsh blah.tcl 
blah.tcl
invalid command name "asdf"
    while executing
"asdf"
    (file "blah.tcl" line 4)
$ openroad -exit blah.tcl 
OpenROAD v2.0-14709-g55cdc9f21 
Features included (+) or not (-): +Charts +GPU +GUI +Python
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
blah.tcl
Error: 
Stack Trace:
invalid command name "asdf"
    while executing
"::unknown {*}$args"
    (procedure "sta_unknown" line 28)
    invoked from within
"asdf"
    (file "blah.tcl" line 4)
$ cat blah.tcl 
puts [info script]


asdf
```